site/src/components/Resources/PortForwardButton.tsx

```diff
@@ -0,0 +1,274 @@
+        id={id}
+        open={isOpen}
+        anchorEl={anchorRef.current}
+        onClose={onClose}
+        anchorOrigin={{
+          vertical: "bottom",
+          horizontal: "left",
+        }}
+        transformOrigin={{
+          vertical: "top",
+          horizontal: "left",
+        }}
+      >
+        <HelpTooltipTitle>Port forward</HelpTooltipTitle>
+        <PortForwardPopoverView {...props} ports={listeningPorts?.ports} />
+      </Popover>
+    </>
+  )
+}
+
+export const PortForwardPopoverView: React.FC<
+  PortForwardButtonProps & { ports?: WorkspaceAgentListeningPort[] }
+> = (props) => {
+  const { host, workspaceName, agentName, username, ports } = props
+
+  return (
+    <>
+      <Box
+        sx={{
+          padding: (theme) => theme.spacing(2.5),
+          borderBottom: (theme) => `1px solid ${theme.palette.divider}`,
+        }}
+      >
+        <HelpTooltipTitle>Ports</HelpTooltipTitle>
+        <HelpTooltipText
+          sx={{ color: (theme) => theme.palette.text.secondary }}
+        >
+          {ports?.length === 0
+            ? "No open ports were detected."
+            : "Port URLs are only accessible by you."}
+        </HelpTooltipText>
+        <Box sx={{ marginTop: (theme) => theme.spacing(1.5) }}>
+          {ports?.map((p) => {
+            const url = portForwardURL(
+              host,
+              p.port,
+              agentName,
+              workspaceName,
+              username,
+            )
+            const label =
+              p.process_name !== "" ? p.process_name : `Port ${p.port}`
+            return (
+              <Link
+                underline="none"
+                sx={{
+                  color: (theme) => theme.palette.text.primary,
+                  fontSize: 14,
+                  display: "flex",
+                  alignItems: "center",
+                  gap: 1,
+                  py: 0.5,
+                  fontWeight: 500,
+                }}
+                key={p.port}
+                href={url}
+                target="_blank"
+                rel="noreferrer"
+              >
+                <OpenInNewOutlined sx={{ width: 14, height: 14 }} />
+                {label}
+                <Box
+                  component="span"
+                  sx={{
+                    ml: "auto",
+                    color: (theme) => theme.palette.text.secondary,
+                    fontSize: 13,
+                    fontWeight: 400,
+                  }}
+                >
+                  {p.port}
+                </Box>
+              </Link>
+            )
+          })}
+        </Box>
+      </Box>
+
+      <Box
+        sx={{
+          padding: (theme) => theme.spacing(2.5),
+        }}
+      >
+        <HelpTooltipTitle>Open app</HelpTooltipTitle>
+        <HelpTooltipText
+          sx={{ color: (theme) => theme.palette.text.secondary }}
+        >
+          Access ports running on the agent with the port, agent name, workspace
```

Comment:
You lost me here :)

Didn't you skip some part of the message? "..as shown below".

<img width="398" alt="Screenshot 2023-08-03 at 18 24 01" src="https://github.com/coder/coder/assets/14044910/07db2260-376f-41b6-b0a9-89e60b35434c">

Reply:
Yes, we removed this copy thing since it was not very helpful tho. Discussion: https://codercom.slack.com/archives/C020UT0HRJB/p1690755533370929

Reply:
We should also remove the extra text then. Let's change the copy to "Access ports running on the agent"